python/lsst/obs/auxTel/auxTelMapper.py

```diff
@@ -0,0 +1,338 @@
+        return self.offsetDate(dateObs, 0.5*exposureTime)
+
+
+def assemble_raw(dataId, componentInfo, cls):
+    """Called by the butler to construct the composite type "raw".
+
+    Note that we still need to define "_raw" and copy various fields over.
+
+    Parameters
+    ----------
+    dataId : `lsst.daf.persistence.dataId.DataId`
+        the data ID
+    componentInfo : `dict`
+        dict containing the components, as defined by the composite definition in the mapper policy
+    cls : 'object'
+        unused
+
+    Returns
+    -------
+    exposure : `lsst.afw.image.exposure.exposure`
+        The assembled exposure
+    """
+    from lsst.ip.isr import AssembleCcdTask
+
+    config = AssembleCcdTask.ConfigClass()
+    config.doTrim = False
+
+    assembleTask = AssembleCcdTask(config=config)
+
+    ampExps = componentInfo['raw_amp'].obj
+    if len(ampExps) == 0:
+        raise RuntimeError("Unable to read raw_amps for %s" % dataId)
+
+    ccd = ampExps[0].getDetector()      # the same (full, CCD-level) Detector is attached to all ampExps
+
+    ampDict = {}
+    for amp, ampExp in zip(ccd, ampExps):
+        ampDict[amp.getName()] = ampExp
+
+    exposure = assembleTask.assembleCcd(ampDict)
+
+    md = componentInfo['raw_hdu'].obj
+    exposure.setMetadata(md)
+    #
+    # We need to standardize, but have no legal way to call std_raw.  The butler should do this for us.
+    #
+    atm = AuxTelMapper()
+    exposure = atm.std_raw(exposure, dataId)
+
+    return exposure
+
+
+class AuxTelMapper(CameraMapper):
+    """The Mapper for the AuxTel."""
+
+    packageName = 'obs_auxTel'
+    MakeRawVisitInfoClass = AuxTelMakeRawVisitInfo
+
+    def __init__(self, inputPolicy=None, **kwargs):
+        """Initialization for the AuxTel Mapper."""
+        policyFile = dafPersist.Policy.defaultPolicyFile(self.packageName, "auxTelMapper.yaml", "policy")
+        policy = dafPersist.Policy(policyFile)
+
+        CameraMapper.__init__(self, policy, os.path.dirname(policyFile), **kwargs)
+        #
+        # The composite objects don't seem to set these
+        #
+        for d in (self.mappings, self.exposures):
+            d['raw'] = d['_raw']
+
+        # self.filterIdMap = {}           # where is this used?  Generating objIds??
+
+        afwImageUtils.defineFilter('NONE', 0.0, alias=['no_filter', 'OPEN', 'empty'])
+        afwImageUtils.defineFilter('275CutOn', 0.0, alias=[])
+        afwImageUtils.defineFilter('550CutOn', 0.0, alias=[])
+        afwImageUtils.defineFilter('green', 0.0, alias=[])
+        afwImageUtils.defineFilter('blue', 0.0, alias=[])
+
+    def _makeCamera(self, policy, repositoryDir):
+        """Make a camera (instance of lsst.afw.cameraGeom.Camera) describing the camera geometry."""
+        return AuxTel()
+
+    def _extractDetectorName(self, dataId):
+        return dataId["ccd"]
+
+    def _computeCcdExposureId(self, dataId):
+        """Compute the 64-bit (long) identifier for a CCD exposure.
```

Comment:
Probably all methods should have parameter definitions in their docstrings.

At this point, the docstring format also changes from sphinx style to doxygen style.  I think it would be good to use sphinx throughout.

Reply:
Oh yeah, gross, I totally agree!